valkey.conf

```diff
@@ -2072,13 +2072,13 @@ stream-node-max-entries 100
 # server is idle the rehashing is never complete and some more memory is used
 # by the hash table.
 #
-# The default is to use this millisecond 10 times every second in order to
-# actively rehash the main dictionaries, freeing memory when possible.
+# The default is to use this millisecond 10 times every second when hz is 10,
+# in order to actively rehash the main dictionaries, freeing memory when possible.
 #
 # If unsure:
 # use "activerehashing no" if you have hard latency requirements and it is
 # not a good thing in your environment that the server can reply from time to time
-# to queries with 2 milliseconds delay.
+# to queries with 2 milliseconds delay when hz is 10.
```

Comment:
Is it really 2ms delay? The rehashing is 1ms plus very little extra I think...

Reply:
Haven't actually measured it, it should be just ms + delay, delay is dictRehash(d, 100) at most i think, so it is 1ms plus vert little extra. Maybe that was the case in the past.

Reply:
Yes, probably in the past. The number 100 in dictRehash(d, 100) was probably chosen to be near one ms. Maybe they measured the time only in milliseconds, not microseconds.

Reply:
Is it actually a good idea to disable active rehashing if the user has hard latency requirements, or is it better to increase the hz?

Maybe we should just delete this "If unsure" explanation?

Reply:
The documentation for `hz` already recommends higher hz for users with latency requirements:

```
# By default "hz" is set to 10. Raising the value will use more CPU when
# the server is idle, but at the same time will make the server more responsive when
# there are many keys expiring at the same time, and timeouts may be
# handled with more precision.
```

I think there is no good reason to turn off active rehashing, just like using synchronous freeing is not a good idea.

Reply:
> Is it actually a good idea to disable active rehashing if the user has hard latency requirements, or is it better to increase the hz?

good point, i'm not sure about it, i don't seem to meet any problems with activerehashing (1% CPU), i think activerehashing is controllable, the other one i think is uncontrollable is passive rehashing (the one when find / add and rehashing), passive rehashing it results in more requests and more rehashing. We have a configuration item to disable it in the internal fork (and always keep activerehashing open)

> Maybe we should just delete this "If unsure" explanation?

yes, i think we can delete this comment

Reply:
> the other one i think is uncontrollable is passive rehashing (the one when find / add and rehashing), passive rehashing it results in more requests and more rehashing

I think we need to rehash one step on `add` to make sure rehashing is finished before we have to do the next resize. We can skip rehashing on `find` though. The problem with avoiding rehashing is that we can have two tables for a very long time, many of them (including hash, zset keys etc.) and they use more memory, but maybe that's OK...